File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> twoSumMap = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            
            int complement = target - nums[i];
            boolean contains = (twoSumMap.containsKey(complement));
            if (contains) {
                    return new int[]{twoSumMap.get(complement), i}; // Return indices of complement and current number
                }
                twoSumMap.put(nums[i], i);
          }

            
        return null; // Return null if no solution is found
    }
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        complement = target
        complementMap = {}
        result=[]
        for idx, num in enumerate(nums):
            complement = target-num #9-2=7
            if complement in complementMap.keys():#if 7 is already a key in complementMap
                idxnumber1 = idx #first number is the number at index 0 in nums array
                idxnumber2 = complementMap[complement] #this returns the index of the complement
                result = [idxnumber1,idxnumber2]
                return result    
            else:
                complementMap[num] = idx #complementMap{2:0} store number and index of number inside nums array in this dict
        